src/execution_engine.rs

```diff
@@ -240,6 +291,22 @@ impl ExecutionEngine {
     }
 }
 
+/// A wrapper around a function pointer which ensures the symbol being pointed
+/// to doesn't accidentally outlive its execution engine.
+#[derive(Debug, Clone)]
+pub struct Symbol<F> {
```

Comment:
Is `Symbol` what LLVM calls it? (I do recall a `LLVMGetSymbol` function) I feel like `Function` or even `RuntimeFunction` might be a more descriptive name